compiler/noirc_frontend/src/signed_field.rs

```diff
@@ -204,4 +259,67 @@ mod tests {
 
         assert!(two > neg_two);
     }
+
+    #[test]
+    fn addition() {
```

Comment:
These tests look good! Could we add some cases with zero to these operators before merging though?

Reply:
I think tests to show we can't construct a `-0` value would be good too. Maybe it should be an assertion when constructing a SignedField as well?

Reply:
Done! Yes, when I added tests for zero what you mentioned started to happen, so I had to add a few `if`s to account for that.